### PR TITLE
Internal Improvements

### PR DESCRIPTION
**Pull request by tgiphil**
+ Additional settings and command line options
+ Minor refactor of exception handling
+ Additional Explorer features

### CODE REVIEW

File: Source/Docs/settings-options.rst [hunk 1]
```diff
@@ -109,6 +109,8 @@ Explorer Settings
    :widths: 50, 200
 
     Explorer.Filter,Specifies the default method filter name when Explorer is launched
+	Explorer.Start,"If true, immediately compiler all methods upon launch"
```
Comment (charsleysa) [inline Source/Docs/settings-options.rst:112]: Should be compile not compiler

File: Source/Mosa.Compiler.Framework/Stages/EnterSSAStage.cs [hunk 2]
```diff
@@ -19,7 +21,7 @@ public sealed class EnterSSAStage : BaseMethodCompilerStage
 	private Dictionary<BasicBlock, SimpleFastDominance> blockAnalysis;
 	private Dictionary<Operand, List<BasicBlock>> assignments;
 	private Dictionary<Operand, Operand> parentOperand;
-	private List<Context> phiInstructions;
+	//private List<Context> phiInstructions;
```
Comment (charsleysa) [inline Source/Mosa.Compiler.Framework/Stages/EnterSSAStage.cs:24]: Commented code

File: Source/Mosa.Compiler.MosaTypeSystem/SignatureComparer.cs [hunk 3]
```diff
@@ -64,14 +64,14 @@ public static int GetHashCode(MosaType type)
 			// Hash code DOES not need to be unique, so to save time ArrayInfo is not hashed
 			case MosaTypeCode.Array:
 				if (type.ElementType == null)
-					throw new InvalidOperationException("Element type of type is null!");
+					throw new InvalidCompilerOperationException("Element type of type is null");
```
Comment (charsleysa) [inline Source/Mosa.Compiler.MosaTypeSystem/SignatureComparer.cs:67]: Inconsistent removal of exclamation marks

Comment (tgiphil) [reply Source/Mosa.Compiler.MosaTypeSystem/SignatureComparer.cs:67]: Found and updated.

File: Source/Docs/settings-options.rst (hunk 1 above)
Comment (tgiphil) [reply Source/Docs/settings-options.rst]: Corrected.

File: Source/Mosa.Compiler.Framework/Stages/EnterSSAStage.cs (hunk 2 above)
Comment (tgiphil) [reply Source/Mosa.Compiler.Framework/Stages/EnterSSAStage.cs:24]: Keeping it for now.

**Review by charsleysa — APPROVED** [review]
LGTM

**Pull request merged**